File: native/Sources/XrayNativeCore/Models/RuntimeTrafficSnapshot.swift
import Foundation

public struct RuntimeTrafficSnapshot: Codable, Equatable, Sendable {
    public let uploadBytesPerSecond: Int
    public let downloadBytesPerSecond: Int

    public init(uploadBytesPerSecond: Int, downloadBytesPerSecond: Int) {
        self.uploadBytesPerSecond = uploadBytesPerSecond
        self.downloadBytesPerSecond = downloadBytesPerSecond
    }

    public static let zero = RuntimeTrafficSnapshot(
        uploadBytesPerSecond: 0,
        downloadBytesPerSecond: 0
    )
}

File: lib/src/core/models/runtime_traffic_snapshot.dart
class RuntimeTrafficSnapshot {
  const RuntimeTrafficSnapshot({
    required this.uploadBytesPerSecond,
    required this.downloadBytesPerSecond,
  });

  const RuntimeTrafficSnapshot.zero()
      : uploadBytesPerSecond = 0,
        downloadBytesPerSecond = 0;

  final int uploadBytesPerSecond;
  final int downloadBytesPerSecond;
}
